### PR TITLE
Add FeedbackFactory implementations for CrashFeedback, CombinedFeedback and DiffFeedback.

### PR DESCRIPTION
**Pull request by djoooooe**
This allows using more complex feedback constructs in `StdTMinMutationalStage` without having to implement `FeedbackFactory` for them.

Example:
```
let diff_feedback = DiffFeedback::new("MyDiffFeedback", &observer_a, &observer_b, |a, b| {
    compare_observations(a, b)
}).unwrap();

let objective = feedback_or!(diff_feedback, CrashFeedback::new());

let min_stage = StdTMinMutationalStage::new(
    mutator,
    objective,
    1 << 10,
);
```

### CODE REVIEW

**Comment by addisoncrump** [comment]
Nice. Can we remove the `type` definitions for the default factories?

**Comment by djoooooe** [comment]
> Nice. Can we remove the `type` definitions for the default factories?

I don't know, maybe someone else can comment on that...

**Comment by addisoncrump** [comment]
Ah, let me rephrase. Please remove the existing `type` declarations (e.g., `type CrashFeedbackFactory`) where you've just made it possible for the Feedbacks to produce themselves.

**Review by addisoncrump — CHANGES_REQUESTED** [review]
Ah, let me rephrase. Please remove the existing `type` declarations (e.g., `type CrashFeedbackFactory`) where you've just made it possible for the Feedbacks to produce themselves.

**Comment by djoooooe** [comment]
I removed `type CrashFeedbackFactory`, I don't see another type made obsolete by my changes.

**Review by addisoncrump — APPROVED** [review]
LGTM
